src/middleware/builtin.ts

```diff
@@ -210,30 +211,40 @@ export function matchConstraints(
  * Middleware that filters out messages that don't match pattern
  */
 export function matchMessage(
-  pattern: string | RegExp,
+  pattern: string | RegExp | (string | RegExp)[],
 ): Middleware<SlackEventMiddlewareArgs<'message' | 'app_mention'>> {
   return async ({ event, context, next }) => {
-    let tempMatches: RegExpMatchArray | null;
-
+    let tempMatches: RegExpMatchArray | null = null;
+    let pattArray: (string | RegExp)[] = [];
     if (!('text' in event) || event.text === undefined) {
       return;
     }
-
-    // Filter out messages or app mentions that don't contain the pattern
-    if (typeof pattern === 'string') {
-      if (!event.text.includes(pattern)) {
+    // Filter out messages or app mentions that don't contain the pattern]
+    if (typeof pattern === 'string' || util.types.isRegExp(pattern)) {
+      pattArray = [pattern];
+    } else {
+      pattArray = pattern;
+    }
+    let matched = false;
+    pattArray.forEach((word) => {
```

Comment:
No need to have the extra parenthesis here.  (I don't know if prettier was the one that added that though).  Also, word doesn't really describe what this is. I would name it pattern.

Reply:
It turns out the parentheses *are* due to Prettier.
We replaced 'word' by the more verbose 'patternArrayItem'.